sources/packages/api/src/services/application/application.service.ts

```diff
@@ -653,36 +655,49 @@ export class ApplicationService extends RecordDataModelService<Application> {
     offering: EducationProgramOffering,
     auditUserId: number,
   ): Promise<Application> {
-    const application = await this.repo
-      .createQueryBuilder("application")
-      .select([
-        "application.id",
-        "currentAssessment.id",
-        "currentAssessment.assessmentWorkflowId",
-      ])
-      .innerJoin("application.currentAssessment", "currentAssessment")
-      .where("application.id = :applicationId", { applicationId })
-      .andWhere("application.location.id = :locationId", { locationId })
-      .andWhere("application.applicationStatus != :applicationStatus", {
-        applicationStatus: ApplicationStatus.overwritten,
-      })
-      .andWhere("application.pirStatus = :pirStatus", {
-        pirStatus: ProgramInfoStatus.required,
-      })
-      .getOne();
-
-    if (!application) {
-      throw new CustomNamedError(
-        "Not able to find an application that requires a PIR to be completed.",
-        PIR_REQUEST_NOT_FOUND_ERROR,
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      const application = await applicationRepo
+        .createQueryBuilder("application")
+        .select([
+          "application.id",
+          "currentAssessment.id",
+          "student.id",
+          "currentAssessment.assessmentWorkflowId",
+        ])
+        .innerJoin("application.currentAssessment", "currentAssessment")
+        .innerJoin("application.student", "student")
+        .where("application.id = :applicationId", { applicationId })
+        .andWhere("application.location.id = :locationId", { locationId })
+        .andWhere("application.applicationStatus != :applicationStatus", {
+          applicationStatus: ApplicationStatus.overwritten,
+        })
+        .andWhere("application.pirStatus = :pirStatus", {
+          pirStatus: ProgramInfoStatus.required,
+        })
+        .getOne();
+
+      if (!application) {
+        throw new CustomNamedError(
+          "Not able to find an application that requires a PIR to be completed.",
+          PIR_REQUEST_NOT_FOUND_ERROR,
+        );
+      }
+      const auditUser = { id: auditUserId } as User;
+      application.currentAssessment.offering = offering;
+      application.currentAssessment.modifier = auditUser;
+      application.pirStatus = ProgramInfoStatus.completed;
+      application.modifier = auditUser;
+      await this.studentRestrictionService.assessSINRestrictionForOfferingId(
```

Comment:
This is the new method, the others changes are due to the addition of the transaction only.